report_common_nrq/models/account_invoice.py

```diff
@@ -19,3 +19,16 @@ class ResPartner(models.Model):
     def onchange_partner(self):
         for invoice in self:
             invoice.delivery_note = invoice.partner_id.delivery_note
+
+    @api.onchange('invoice_line_ids')
+    def _onchange_origin(self):
+        res = super(ResPartner, self)._onchange_origin()
+        purchase_ids = self.invoice_line_ids.mapped('purchase_id')
+        self.doc_title = ""
+        if purchase_ids:
+            for title in purchase_ids.mapped('doc_title'):
+                if title != False:
+                    if self.doc_title == "":
+                        self.doc_title += title
+                    else:
+                        self.doc_title += ", " + title
```

Comment:
I'd rather override the method without using super for this case.  Something like this:
```
     @api.onchange('invoice_line_ids')
     def _onchange_origin(self):
        purchase_ids = self.invoice_line_ids.mapped('purchase_id')
        if purchase_ids:
            self.origin = ', '.join(purchase_ids.mapped('name'))
            doc_title = ""
            for title in purchase_ids.mapped('doc_title'):
                if title:
                    if not doc_title:
                        doc_title = title
                    else:
                        doc_title += ", " + title
            self.doc_title = doc_title
```